torchgeo/datasets/geo.py

```diff
@@ -72,7 +72,7 @@ class GeoDataset(Dataset[dict[str, Any]], abc.ABC):
        dataset = landsat7 | landsat8
     """
 
-    paths: Union[str, Iterable[str]]
+    paths: Union[str, Iterable[str], os.PathLike, Iterable[os.PathLike]]
```

Comment:
Let's add a type alias to make things simpler. In `torchgeo/datasets/utils.py`, you can add:
```python
import os
from typing import Union, TypeAlias

Path: TypeAlias = Union[str, bytes, os.PathLike[str], os.PathLike[bytes]]
```
Then in all other files you can simply replace `str` with `Path` where needed. This will be less code duplication.

Reply:
Thank you for your feedback, I will work towards implementing this.